Sources/MuxUploadSDK/PublicAPI/PhotosKit+MuxUpload/PHAsset+MuxUpload.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  MuxUpload+PhotosKit.swift
+//
+
+import Foundation
+import Photos
+
+extension PHAsset {
+
+    /// Convenience method that requests an ``AVAsset``
+    /// containing audiovisual media associated with the callee
+    /// ``PHAsset``. If the request succeeds ``MuxUpload``
+    /// is initialized with the received ``AVAsset``
+    ///
+    /// A ``MuxUpload`` can only be initialized from a ``PHAsset``
+    /// whose ``PHAsset.mediaType`` is ``PHAssetMediaType.video``
+    ///
+    /// - Parameters:
+    ///    - imageManager: an object that facilitates retrieval
+    ///    of asset data associated with the callee
+    ///    - requestOptions: options used when requesting
+    ///    an ``AVAsset`` from a ``PHImageManager`
+    ///    - options: options used to control the direct
+    ///    upload of the input to Mux
+    ///    - uploadURL: the URL of the direct upload that's
+    ///    included in the create a new direct upload URL
+    ///    [response](https://docs.mux.com/api-reference#video/operation/create-direct-upload)
+    ///    - completion: called when initialized ``MuxUpload``
+    ///    is ready, receives nil if the asset data request
+    ///    failed or if the ``PHAsset`` callee is not a video
+    func prepareForDirectUpload(
+        from imageManager: PHImageManager = .default(),
+        requestOptions: PHVideoRequestOptions,
```

Comment:
I agree that this is the right set of request options to use for this. I'd even go a step further and would propose hardcoding the options to these for the time being.

Re: `PHImageRequestOptionsDeliveryMode` / `deliveryMode`, I think it shouldn't affect if the SDK exports a new `AVAsset` or not. To allow it to so creates potential foot guns:

- It exposes implementation details of how input standardization works.
- `deliveryMode` can in some cases make client intent appear ambiguous to the SDK (i.e. how do we handle a possible scenario where the `deliveryMode` is `fastFormat` but the asset is a very high quality one? Upload a reduced quality version of an otherwise high-quality asset? What if it conflicts with the maximum resolution?)
- Would require the SDK to track whether it needs to re-export the input or not, and handle other edge cases it otherwise doesn't need to.

The SDK behavior initially around this should lean towards being too simple, so my vote is always request the highest quality version. Clients can constrain the quality of the upload via `maximumResolution`. There is value in potentially offering a greater level of control after some requests or bug reports come in and we have case studies / use cases to inform the API design.

Reply:
https://github.com/muxinc/swift-upload-sdk/pull/17/commits/31a18e1fda66fc09f2721d37cd8baae39ed976a8